### DISCUSSION TITLE
Using _setPositionDimensions({}) with Paths

### DISCUSSION THREAD

**Discussion by Nineswiss** [fabricjs/fabric.js#7559] [General]
I was able to port the Custom Controls demo project to edit paths and everything works. However when I try to set the new dimensions i get 'fabricObject._setPositionDimensions is not a function'. I can set and move points but get stuck here. My modified code:

`  pathAnchorWrapper(anchorIndex, fn) {
        return function (eventData, transform, x, y) {
            var fabricObject = transform.target,
                absolutePoint = fabric.util.transformPoint({
                    x: (fabricObject.path[anchorIndex][1] - fabricObject.pathOffset.x),
                    y: (fabricObject.path[anchorIndex][2] - fabricObject.pathOffset.y),
                }, 
                fabricObject.calcTransformMatrix()),
                actionPerformed = fn(eventData, transform, x, y),
                newDim = fabricObject._setPositionDimensions({}),
                polygonBaseSize = fabricObject._getNonTransformedDimensions(),
                newX = (fabricObject.path[anchorIndex][1] - fabricObject.pathOffset.x) / polygonBaseSize.x,
                newY = (fabricObject.path[anchorIndex][2] - fabricObject.pathOffset.y) / polygonBaseSize.y;
                fabricObject.setPositionByOrigin(absolutePoint, newX + 0.5, newY + 0.5);
                return actionPerformed;
        }
    }`

**Comment by asturur** [comment]
Yes, the path uses the polyline method without inheriting from it, definitly a dirty trick.

you can call
```
fabric.Polyline.prototype._setPositionDimensions.call(yourPath, options);
```

**Reply by ShaMan123** [reply]
#7841

**Comment by rolitter** [comment]
FabricJS modifyPath
https://codepen.io/rolitter/pen/mdqZWPZ

**Reply by vimlesh1975** [reply]
This is very usefull and i have included this code in my application.
but setting path property as strokeUniform: true, and scaling the path , path shifts while dragging controls.
 https://codepen.io/vimlesh1975/pen/eYydGjX

**Reply by Nineswiss** [reply]
Yup, scaling and then editing the path causes the whole path to move.

**Reply by ShaMan123** [reply]
This is a known issue regarding `strokeUniform`
#6705 #7639

**Reply by vimlesh1975** [reply]
For now it is not a big problem for me. Thanks for replying.

**Comment by ShaMan123** [comment]
[`Path._setPath`](https://github.com/fabricjs/fabric.js/blob/b2d6dd9b2fe27f031973c1e42578d33df9a15b52/src/shapes/path.class.js#L62)
